tensorflow_addons/metrics/tests/matthews_correlation_coefficient_test.py

```diff
@@ -36,30 +37,59 @@ def check_results(obj, value):
 
 
 def test_binary_classes():
-    gt_label = tf.constant([[1.0], [1.0], [1.0], [0.0]], dtype=tf.float32)
-    preds = tf.constant([[1.0], [0.0], [1.0], [1.0]], dtype=tf.float32)
+    gt_label = tf.constant(
+        [[0.0, 1.0], [0.0, 1.0], [0.0, 1.0], [1.0, 0.0]], dtype=tf.float32
+    )
+    preds = tf.constant(
+        [[0.0, 1.0], [1.0, 0.0], [0.0, 1.0], [0.0, 1.0]], dtype=tf.float32
+    )
     # Initialize
-    mcc = MatthewsCorrelationCoefficient(1)
+    mcc = MatthewsCorrelationCoefficient(2)
```

Comment:
Unfortunately this PR breaks this test and the use of MCC with `num_classes=2`. Is this breaking change intended?

Reply:
Let's continue this discussion on your PR.